api/zulip/__init__.py

```diff
@@ -290,16 +290,20 @@ def get_user_agent(self):
                 vendor_version=vendor_version,
                 )
 
-    def do_api_query(self, orig_request, url, method="POST", longpolling=False):
-        # type: (Mapping[str, Any], str, str, bool) -> Dict[str, Any]
+    def do_api_query(self, orig_request, url, files=[], method="POST", longpolling=False):
```

Comment:
Actually, you should also say `files=None` here, and then inside the function...

~~~ py
if files isNone:
    files = []
~~~

Python's semantics for initializing parameters is that you are actually assigning an empty array to `files`, and this array will actually grow every time you call the function!